### PR TITLE
sympy 1.7 fixes

### PR DESCRIPTION
**Pull request by MauriceHendrix**
## Description

- Fixes for sympy 1.7: 
    - Fixed the printer test (eval=false no longer exists but evaluate=false can still be used). 
    - Removed the positional arguments in Variable as these were causing issues, instead use named arguments for name and units. Added aditional test that creates new Variable using sympy solveset

## Motivation and Context

Codegen crashes when trying to use sympy 1.7

## Types of changes

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Documentation checklist

- [X] I have updated all documentation in the code where necessary.
- [X] I have checked spelling in all (new) comments and documentation.
- [X] I have added a note to RELEASE.md if relevant (new feature, breaking change, or notable bug fix).

## Testing
- [x] Testing is done automatically and codecov shows test coverage
- [ ] This cannot be tested automatically 

I ran the test suite with sympy 1.4, 1.5, 1.6, 1.6.2, 1.7 and 1.7.1 (all passing)

### CODE REVIEW

File: cellmlmanip/model.py [hunk 1]
```diff
@@ -1091,12 +1091,12 @@ class Variable(sympy.Dummy):
...
     # Sympy annoyingly overwrites __new__
-    def __new__(cls, name, *args, **kwargs):
-        return super().__new__(cls, name, real=True)
+    def __new__(cls, *args, **kwargs):
+        return super().__new__(cls, real=True)
 
     def __init__(self,
-                 name,
-                 units,
+                 name=None,
```
Comment (MichaelClerx) [inline cellmlmanip/model.py:1098]: Is this still compatible with older versions?

Comment (MauriceHendrix) [reply cellmlmanip/model.py:1098]: I think so, I mean it passed all the tests that use Variable, I didn't change any of those...

**Comment by MichaelClerx** [comment]
I thought the tests weren't running anymore 😆

**Review by MichaelClerx — APPROVED** [review]

**Comment by MauriceHendrix** [comment]
> I thought the tests weren't running anymore 😆

Well appveyor still works on cellmlmanip and of course locally

**Review by MichaelClerx — APPROVED** [review]
OK! I've tried with sympy 1.6.2 too and seems to work for me

**Pull request merged**